### DISCUSSION TITLE
Make the nvm list can detect the symlinks in the nvm-windows root directory

### DISCUSSION THREAD

**Discussion by Humbunklung** [coreybutler/nvm-windows#1030] [General]
The nodejs installed by Visual Studio work load, cannot be added to the nvm list, especially the one is not in the environment variable path. I can make symlinks in the root directory link the origin node path created by visual studio. For example, v16.14.0 links to "C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VisualStudio\NodeJs".
![image](https://github.com/coreybutler/nvm-windows/assets/7733197/e9af82b1-17bd-4ccd-ab80-4e455cdf9be3)

However, the command "nvm list" can not list the node's symlinks added by me, because in the file node.go,  `files[i].IsDir() `will return false, while files[i] is a symlink.
![image](https://github.com/coreybutler/nvm-windows/assets/7733197/c6b92911-ccd3-4baa-9c9c-3c92779762f5)
I modified the function `GetInstalled` like this:
```
func GetInstalled(root string) []string {
	list := make([]semver.Version, 0)
	files, _ := ioutil.ReadDir(root)

	for i := len(files) - 1; i >= 0; i-- {
		if files[i].IsDir() || (files[i].Mode()&os.ModeSymlink == os.ModeSymlink) {
			isnode, _ := regexp.MatchString("v", files[i].Name())

			if isnode {
				currentVersionString := strings.Replace(files[i].Name(), "v", "", 1)
				currentVersion, _ := semver.Make(currentVersionString)

				list = append(list, currentVersion)
			}
		}
	}

	semver.Sort(list)

	loggableList := make([]string, 0)

	for _, version := range list {
		loggableList = append(loggableList, "v"+version.String())
	}

	loggableList = reverseStringArray(loggableList)

	return loggableList
}
```
Then my problem solved, the symlinks are detected.
![image](https://github.com/coreybutler/nvm-windows/assets/7733197/f6584776-61e3-41d0-bf41-feeb54fd985e)

**Comment by coreybutler** [comment]
I've converted this to an issue (#1031). If you'd like to submit your code as a PR against that issue, I can test/merge it.

**Reply by Humbunklung** [reply]
OK.
